custom_components/hueremote/remote.py

```diff
@@ -1,71 +1,109 @@
-        HueRemote,
-        REMOTE_MODELS,
-        async_add_entities,
-        config.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL),
-    )
-
-
-class HueRemote(HueSensorBaseDevice, RemoteDevice):
-    """Class to hold Hue Remote basic info."""
-
-    @property
-    def state(self):
-        """Return the state of the remote."""
-        return self.sensor_data["state"]
-
-    @property
-    def icon(self):
-        """Icon to use in the frontend, if any."""
-        data = self.sensor_data
-        if data:
-            icon = REMOTE_ICONS.get(data["model"])
-            if icon:
-                return icon
-        return "mdi:remote"  # pragma: no cover
-
-    @property
-    def force_update(self):
-        """Force update."""
-        return True
-
-    def turn_on(self, **kwargs):
-        """Do nothing."""
-
-    def turn_off(self, **kwargs):
-        """Do nothing."""
+    for bridge_entry_id, bridge in hass.data[HUE_DOMAIN].items():
+
+        @callback
+        def _add_remotes():
+            """Check for new devices to be added to HA."""
+            new_remotes = []
+            api = bridge.api.sensors
+            for item_id in api:
+                sensor = api[item_id]
+                if sensor.type not in _IMPLEMENTED_REMOTE_TYPES:
+                    continue  # pragma: no cover
+
+                existing = bridge.sensor_manager.current.get(sensor.uniqueid)
+                if existing is not None:
+                    continue
+
+                sensor_config = SENSOR_CONFIG_MAP.get(sensor.type)
+                base_name = sensor.name
+                name = sensor_config["name_format"].format(base_name)
+
+                new_remote = sensor_config["class"](sensor, name, bridge)
+                bridge.sensor_manager.current[sensor.uniqueid] = new_remote
+                new_remotes.append(new_remote)
+
+                _LOGGER.debug(
+                    "Setup remote %s: %s", sensor.type, sensor.uniqueid,
+                )
+            if new_remotes:
+                async_add_entities(new_remotes)
+
+        # Add listener to add discovered remotes
+        bridge.sensor_manager.coordinator.async_add_listener(_add_remotes)
+        _add_remotes()
+
+        # Set up updates at scan_interval
+        async def _update_remotes(now=None):
+            """Request a bridge data refresh to update states on remotes."""
+            await bridge.sensor_manager.coordinator.async_refresh()
+            _LOGGER.debug("Update requested at %s", now)
+
+        remote_sc = config.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL)
+        if remote_sc < bridge.sensor_manager.coordinator.update_interval:
+            # Add listener to update remotes at high rate
+            # TODO change sm.coordinator.update_interval instead.
+            #  It can be done right here, but any bridge.reset event
+            #  would re-write it, so we maintain our own scheduler meanwhile
+            async_track_time_interval(
```

Comment:
Here I had some doubts about how to proceed.

* The CC is calling `await bridge.sensor_manager.coordinator.async_refresh()` each HA tick, so it is **making a real call**. That one is adding to the one that the HA sensors do, so we have a 1Hz request + 0.2Hz request. It's no big deal, but every 5 seconds there is a repeated call (but it may be debounced).
* Another approach is to call `.async_request_refresh()` instead, which is going to be blocked if a previous call is timing out (see the conversation in https://github.com/home-assistant/core/pull/32751), so as this is a CC, I prefer to ensure quick response and avoid unavailability issues (and warn about in docs).
* And the simpler approach is to just change the `update_interval` of the hue integration to 1Hz, and forget about our updater, but then we have again the possibility of availability issues. And also, every time the bridge is reset (if you disable some entity in the UI, p.ex.), this code won't run again and we lost our 1Hz (an HA restart is needed to recover).

So, in the end, I chose to maintain our own 1Hz forced updater, as IMO is the better option to minimize issues and maximize responsiveness :)

Reply:
`minimize issues and maximize responsiveness` is good by me! Yes we don't want an approach that will fail if the bridge is reset